lib/core/intrinsics/external_intrinsics.cpp

```diff
@@ -0,0 +1,123 @@
+#include "core/intrinsics/external_intrinsics.h"
+
+#include <clang/Frontend/CompilerInstance.h>
+#include "core/transpiler_session/transpiler_session.h"
+#include "util/string_utils.hpp"
+
+#include <algorithm>
+
+namespace oklt {
+
+using namespace llvm;
+namespace fs = std::filesystem;
+
+tl::expected<fs::path, std::string>
+getIntrincisImplSourcePath(TargetBackend backend,
```

Comment:
Yet another `TargetBackend` dependent switch.
It would be nice to have a proper const registry list for this kind of stuff..

Reply:
what does the registery solve in this case ?

Reply:
Simple.
Backend registration is all over the place.
TargetBackend enum, backendFromString, backendToString, isHostCategory, isDeviceCategory, and now getIntrincisImplSourcePath (totally separate).

That's why I'd like to ask for enhancement and consolidate all of them together.

Reply:
Yeap. All the functions you mentioned are serated ones. Do you want to group it only by the fact of switch statement ?

Reply:
I was thinking of using a constexpr initializer list of structures, where each field represents one of the above options, including `intrinsic` and it's path.
Or we could do like LLVM/CLang does using macro auto generators.

In any case, the idea is to aggregate them together into a single place.

Reply:
Ok. Put it into the single place but functionality will be different. What is the reason ? I think It's overcomplexity for nothing

Reply:
> Ok. Put it into the single place but functionality will be different. What is the reason ? I think It's overcomplexity for nothing

At the current state, it's fairly easy to miss `getIntrincisImplSourcePath` function when adding a new backend.
That is why I suggest improving it by introducing a static initializer list of structures inside 'core/target_backends'.
Each structure field represents all the necessary parts for the backend to be registered.

If an author is `oh so lazy` to do that, then please move the function `getIntrincisImplSourcePath` to core/target_backends' so that it does not introduce additional dependencies.